frontend/apps/reader/modules/readerwebsearch.lua

```diff
@@ -0,0 +1,310 @@
+            ["suffix"] = "",
+            ["format"] = "UrlEncode",
+            ["title"] = "ResearchGate",
+        },
+        [11] = {
+            ["active"] = true,
+            ["prefix"] = "https://fr.wikisource.org/w/index.php?search=",
+            ["suffix"] = "",
+            ["format"] = "UrlEncode",
+            ["title"] = "Wikisource",
+        },
+        [12] = {
+            ["active"] = true,
+            ["prefix"] = "https://book4you.org/s/",
+            ["suffix"] = "",
+            ["format"] = "UrlEncode",
+            ["title"] = "ZLibrary",
+        }
+      }
+      G_reader_settings:saveSetting("websearchsites", websearchsites)
+    end
+        
+    title_table = {}
+    for websearchsite, websitedetails in pairs (saved_websearchsites) do
+        table.insert(title_table, websitedetails.title)
+    end
+    table.sort(title_table)
+
+    websearchsites= {}
+    for _, atitle in pairs(title_table) do
+      for _, websearchsite in pairs(saved_websearchsites) do
+        if (atitle == websearchsite.title) then
+          table.insert(websearchsites, websearchsite)
+        end
+      end
+    end
+
+    logger.info("Ordered Web Search Sites:", websearchsites)
+    
+    active_websearchsites = {}
+    for website, data in pairs (websearchsites) do
+    --  table.insert(self.websearchsitetitles, data.title)
+      active_websearchsites[website] = data.active
+    end
+    
+    default_websearchsite = G_reader_settings:readSetting("default_websearchsite")
+    
+    if not default_websearchsite then 
+        default_websearchsite = 1
+        G_reader_settings:saveSetting("default_websearchsite", default_websearchsite)
+    end
+    
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function ReaderWebSearch:genActiveWebSearchSubItem(websearchsite)
+    return {
+        text = self:getWebSearchSiteTitle(websearchsite), 
+        checked_func = function()
+            return ReaderWebSearch:isWebSearchSiteActive(websearchsite)
+        end,
+        callback = function()
+            ReaderWebSearch:toggleWebSearchSiteActive(websearchsite)
+        end
+    }
+end
+
+function ReaderWebSearch:genDefaultWebSearchSubItem(websearchsite)
+    return {
+        text = self:getWebSearchSiteTitle(websearchsite), 
+        checked_func = function()
+            return (self:isWebSearchSiteDefault(websearchsite))
+        end,
+        callback = function()
+            self:setWebSearchSiteDefault(websearchsite)
+        end
+    }
+end
+
+function ReaderWebSearch:addToMainMenu(menu_items)
+    sub_item_table_websearch = {}
+    sub_item_table_defaultwebsearch = {}
+    
+    for websearchsite, websitedetails in pairs (websearchsites) do
+      table.insert(sub_item_table_websearch, self:genActiveWebSearchSubItem(websearchsite))
+      table.insert(sub_item_table_defaultwebsearch, self:genDefaultWebSearchSubItem(websearchsite))
+    end
+    
+    menu_items.websearch = {
+      text = _("Select Active Web Search Sites"),
```

Comment:
This seems rather long, so mainly leaving this as a marker for me or others later.
```suggestion
      text = _("Select active web search sites"),
```